### ISSUE TITLE
Undo after inserting keys and pressing Ctrl+u is broken

### ISSUE THREAD

**Issue by mhchen** [VSCodeVim/Vim#4913]
**Describe the bug**
In insert mode, when I type and then press Ctrl+u (to delete to beginning of line), then try to undo, it only moves my cursor without bringing back the deleted text.

**To Reproduce**

1.  Enter insert mode in the middle of a line
2.  Type some text
3.  `<C-u>`
4. Exit insert mode
5. `Press u`

**Expected behavior**
Pre-existing text (prior to inserting characters) is ignored.

**Screenshots**
![Kapture 2020-05-28 at 23 07 09](https://user-images.githubusercontent.com/1325549/83227085-32c17080-a138-11ea-8af5-227c1e1addc9.gif)

**Environment (please complete the following information):**

- Extension (VsCodeVim) version: 1.14.5
- VSCode version: 1.46.0-insiders
- OS: OS X 10.15.4

**Comment by kevinw** [comment]
Undo and insert mode is super broken! This makes the editor almost unusable for me daily.

**Comment by sethlivingston** [comment]
I'm so thankful for this plugin, but if I fat-finger a few keys and accidentally delete some characters and jump somewhere, then I'm hosed because undo doesn't work. This is a big problem for me.

**Comment by bitshiftnetau** [comment]
I agree with @kevinw I have just turned this plugin off after the broken undo feature undid a whole swathe of text, rather than just one character at a time. Unusable.

**Comment by berknam** [comment]
@bitshiftnetau  Were you by any chance recording a macro before that resulted in error?
